Libplanet/Blocks/InvalidBlockBytesLengthException.cs

```diff
@@ -0,0 +1,49 @@
+#nullable enable
+using System;
+using Libplanet.Blockchain.Policies;
+
+namespace Libplanet.Blocks
+{
+    /// <summary>
+    /// The exception that is thrown when a <see cref="Block{T}"/>'s
+    /// <see cref="Block{T}.BytesLength"/> (i.e., the length of bytes in its serialized format)
+    /// is too long.
+    /// </summary>
+    [Serializable]
+    public sealed class InvalidBlockBytesLengthException : InvalidBlockException
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="InvalidBlockBytesLengthException"/> class.
+        /// </summary>
+        /// <param name="blockBytesLength">The invalid <see cref="Block{T}"/>'s
+        /// <see cref="Block{T}.BytesLength"/>.  It is automatically included to the
+        /// <see cref="Exception.Message"/> string.</param>
+        /// <param name="maxBlockBytesLength">The maximum allowed length of bytes.  It is
+        /// automatically included to the <see cref="Exception.Message"/> string.</param>
+        /// <param name="message">The message that describes the error.</param>
+        public InvalidBlockBytesLengthException(
+            int blockBytesLength,
+            int maxBlockBytesLength,
+            string message
+        )
+            : base(
+                $"{message}\n" +
+                $"Actual: #{blockBytesLength} bytes\n" +
+                $"Allowed (max): #{maxBlockBytesLength}")
+        {
+            BlockBytesLength = blockBytesLength;
+            MaxBlockBytesLength = maxBlockBytesLength;
+        }
+
+        /// <summary>
+        /// The bytes length of the actual block.
+        /// </summary>
+        public int BlockBytesLength { get; set; }
```

Comment:
I guess we need to add codes about serialization for these fields to preserve them.

Reply:
Added.